### PR TITLE
Automated ER diagram generation

### PR DESCRIPTION
**Pull request by Scilly-guy**
Added make command to refresh entity relationship diagram.

### CODE REVIEW

**Comment by Scilly-guy** [comment]
Suggest updating read me to include instructions to install graphviz to enable ER diagram generation.

File: insert-diagram.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        print(f"Updated HTML file has been saved to {output_file}.")
+    except Exception as e:
+        print(f"An error occurred: {e}")
+
+
+# Example usage:
+# replace_placeholder('index.html', 'graphic.svg', 'updated_index.html')
+
+# If running from the command line:
+if __name__ == "__main__":
```
Comment (grundleborg) [inline insert-diagram.py:30]: Is this file written by you or copy-pasted from a template/example/docs somewhere? If the latter we need to check the licence and add appropriate licence header if necessary.

File: template_entity_relationship_diagram.html [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  viewBox.width = newWidth;
+  viewBox.height = newHeight;
+  
+  svg.setAttribute("viewBox", `${viewBox.x} ${viewBox.y} ${viewBox.width} ${viewBox.height}`);
+  scale=1;
+});
+
+    </script>
+</body>
+</html>
```
Comment (grundleborg) [inline template_entity_relationship_diagram.html:92]: As with the python file, just want to check this as well: Is this file written by you or copy-pasted from a template/example/docs somewhere? If the latter we need to check the licence and add appropriate licence header if necessary.

**Review by grundleborg — CHANGES_REQUESTED** [review]
This looks good. I have a couple of additional requests:
1. Please can you add instructions (and a note on additional dependencies) to the README.md file.
2. Please can you add all the files generated by this script to `.gitignore` so that they don't get accidentally checked in to the repo.
3. To make the above simpler and keep things tidier, I think you should create a subdirectory for the entity relationship diagram stuff, and move the `insert-diagram.py` and `template_entity_relatoinship_diagram.html` files into it, as well as adjusting the generation commands to create all their files within that directory too.
4. You've added a dependency which means you'll need to run `poetry lock` (I think) to update `poetry.lock` file.

File: insert-diagram.py (hunk 1 above)
Comment (Scilly-guy) [reply insert-diagram.py:30]: It is written by chatgpt and checked by me.

File: template_entity_relationship_diagram.html (hunk 2 above)
Comment (Scilly-guy) [reply template_entity_relationship_diagram.html:92]: This is written by me.

**Review by grundleborg — APPROVED** [review]
Fixed a couple of minor things when testing on my computer, but this LGTM now.

**Pull request merged**